src/app/_service/offer.service.ts

```diff
@@ -0,0 +1,25 @@
+import {Injectable} from "@angular/core";
+import {HttpClient} from "@angular/common/http";
+import {AuthenticationService} from "./authentication.service";
+
+export class Offer {
+  constructor(
+    public name: string,
+    public url: string,
+    public price: number,
+  ) {
+  }
+}
+
+@Injectable({providedIn: 'root'})
+export class OfferService {
+
+  constructor(private httpClient: HttpClient,
+              private authenticationService: AuthenticationService,) {
+  }
+
+  getOffers() {
+    const headers = this.authenticationService.getHeaders();
+    return this.httpClient.get<Array<Offer>>('http://localhost:8080/api/allegro/offers', {headers})
```

Comment:
Maybe a config file to keep such addresses? Or a variable? Just to find it easier when URL changes.